Formula/astro.rb

```diff
@@ -0,0 +1,19 @@
+class Astro < Formula
+  desc "To build and run Airflow DAGs locally and interact with the Astronomer API"
+  homepage "https://www.astronomer.io/"
+  url "https://github.com/astronomer/astro-cli/archive/refs/tags/v1.0.2.tar.gz"
+  sha256 "5121b7697f7db7a85b4e35af2410cb4832fa0fb3983217e3168f3cdbc2ef72b1"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+    system "go", "build", *std_go_args(ldflags: "-s -w -X github.com/astronomer/astro-cli/version.CurrVersion=#{version}")
+  end
+
+  test do
+    run_output = shell_output("#{bin}/astro version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.